sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -320,11 +320,13 @@ Command *Scheduler::GraphBuilder::addHostAccessor(Requirement *Req) {
 
   AllocaCommandBase *SrcAllocaCmd = nullptr;
 
-  if (Record->MAllocaCommands.empty())
+  if (Record->MAllocaCommands.empty() || IsSuitableSubReq(Req))
     SrcAllocaCmd = getOrCreateAllocaForReq(Record, Req, HostQueue);
```

Comment:
getOrCreateAkkicaForReq() method contains a proper procedure to create/find alloca cmd for subbuffer. That is why there is this kind of hack: regardless of whether alloca commands collection is non-empty go on and getOrCreate it. The method will lookup for it in first place, though. It'll not find the correct alloca command 'cause there is no equality of MOffsetInBytes. After that the method will create a proper alloca cmd for subbuffer.